zipkin-core/src/main/java/com/smoketurner/dropwizard/zipkin/RabbitMQZipkinFactory.java

```diff
@@ -0,0 +1,156 @@
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.smoketurner.dropwizard.zipkin.managed.ReporterManager;
+import com.smoketurner.dropwizard.zipkin.metrics.DropwizardReporterMetrics;
+import brave.http.HttpTracing;
+import io.dropwizard.setup.Environment;
+import io.dropwizard.util.Duration;
+import io.dropwizard.validation.MinDuration;
+import zipkin2.Span;
+import zipkin2.reporter.AsyncReporter;
+import zipkin2.reporter.ReporterMetrics;
+import zipkin2.reporter.amqp.RabbitMQSender;
+
+@JsonTypeName("amqp")
+public class RabbitMQZipkinFactory extends AbstractZipkinFactory {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(RabbitMQZipkinFactory.class);
+
+    @NotEmpty
+    private String addresses = "";
+
+    @NotEmpty
+    private String queue = "zipkin";
+
+    @NotEmpty
+    private String username = "guest";
+
+    @NotEmpty
+    private String password = "guest";
+
+    @NotEmpty
+    private String virtualHost = "/";
+
+    @NotNull
+    @MinDuration(value = 1, unit = TimeUnit.MILLISECONDS)
+    private Duration connectionTimeout = Duration.seconds(60);
+
+    @JsonProperty
+    public String getAddresses() {
+        return addresses;
+    }
+
+    @JsonProperty
+    public void setAddresses(String addresses) {
+        this.addresses = addresses;
+    }
+
+    @JsonProperty
+    public String getQueue() {
+        return queue;
+    }
+
+    @JsonProperty
+    public void setQueue(String queue) {
+        this.queue = queue;
+    }
+
+    @JsonProperty
+    public String getUsername() {
+        return username;
+    }
+
+    @JsonProperty
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    @JsonProperty
+    public String getPassword() {
+        return password;
+    }
+
+    @JsonProperty
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    @JsonProperty
+    public String getVirtualHost() {
+        return virtualHost;
+    }
+
+    @JsonProperty
+    public void setVirtualHost(String virtualHost) {
+        this.virtualHost = virtualHost;
+    }
+
+    @JsonProperty
+    public Duration getConnectionTimeout() {
+        return connectionTimeout;
+    }
+
+    @JsonProperty
+    public void setConnectionTimeout(Duration timeout) {
+        this.connectionTimeout = timeout;
+    }
+
+    /**
+     * Build a new {@link HttpTracing} instance for interfacing with Zipkin
+     *
+     * @param environment
+     *            Environment
+     * @return Brave instance
+     */
+    @Override
+    public Optional<HttpTracing> build(@Nonnull final Environment environment) {
+        if (!isEnabled()) {
+            LOGGER.warn("Zipkin tracing is disabled");
+            return Optional.empty();
+        }
+
+        final ReporterMetrics metricsHandler = new DropwizardReporterMetrics(
+                environment.metrics());
+
+        final RabbitMQSender sender = RabbitMQSender.newBuilder()
```

Comment:
probably should close these things.. I thought there was a spot which did lifecycle somewhere already?

Reply:
that happens inside the managers - https://github.com/smoketurner/dropwizard-zipkin/blob/master/zipkin-core/src/main/java/com/smoketurner/dropwizard/zipkin/managed/ReporterManager.java#L65